projects/pathfinder-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/DetailService.kt

```diff
@@ -0,0 +1,66 @@
+package uk.gov.justice.digital.hmpps.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.controller.IdentifierType
+import uk.gov.justice.digital.hmpps.entity.ConvictionEventRepository
+import uk.gov.justice.digital.hmpps.entity.CustodyRepository
+import uk.gov.justice.digital.hmpps.entity.DetailReleaseRepository
+import uk.gov.justice.digital.hmpps.entity.DetailRepository
+import uk.gov.justice.digital.hmpps.entity.findByCrn
+import uk.gov.justice.digital.hmpps.entity.findByNomsNumber
+import uk.gov.justice.digital.hmpps.model.Detail
+import uk.gov.justice.digital.hmpps.model.KeyDate
+import uk.gov.justice.digital.hmpps.model.Name
+import uk.gov.justice.digital.hmpps.model.name
+import java.time.LocalDate
+
+@Service
+class DetailService(
+    private val detailRepository: DetailRepository,
+    private val convictionEventRepository: ConvictionEventRepository,
+    private val custodyRepository: CustodyRepository,
+    private val detailReleaseRepository: DetailReleaseRepository
+) {
+    fun getDetails(value: String, type: IdentifierType): Detail {
+        val p = when (type) {
+            IdentifierType.CRN -> detailRepository.findByCrn(value)
+            IdentifierType.NOMS -> detailRepository.findByNomsNumber(value)
+        }
+        val c = convictionEventRepository.getAllByConvictionEventPersonId(p.id)
+        var mainOffence: String? = ""
```

Comment:
This is true and I've updated the code to assume the main offence is always there. However the model returned may not have a main offence - since (although unlikely) the event may not have been created yet. Pathfinder scoops up data from Nomis and Delius and merges it together. If the data is not present it shouldn't fail... at least I don't think it should, so I've perhaps been a little more lenient.